apps/ledger-live-mobile/src/families/ethereum/EditTransaction/MethodSelection.tsx

```diff
@@ -0,0 +1,226 @@
+  );
+
+  let isOldestEditableOperation = true;
+
+  account.pendingOperations.forEach(pendingOperation => {
+    if (isEditableOperation(account, pendingOperation)) {
+      if (
+        operation.transactionSequenceNumber &&
+        pendingOperation.transactionSequenceNumber &&
+        pendingOperation.transactionSequenceNumber <
+          operation.transactionSequenceNumber
+      ) {
+        isOldestEditableOperation = false;
+      }
+    }
+  });
+
+  const currency = getAccountCurrency(account);
+
+  const bridge = getAccountBridge(account, parentAccount as Account);
+
+  const onSelect = (option: "cancel" | "speedup") => {
+    switch (option) {
+      case "cancel":
+        transactionToEdit.amount = new BigNumber(0);
+        transactionToEdit.data = Buffer.from("");
+        transactionToEdit.nonce = operation.transactionSequenceNumber;
+        transactionToEdit.allowZeroAmount = true;
+        transaction.mode = "send";
+
+        if (EIP1559ShouldBeUsed(currency)) {
+          if (transactionToEdit.maxPriorityFeePerGas) {
+            transactionToEdit.maxPriorityFeePerGas = new BigNumber(
+              transactionToEdit.maxPriorityFeePerGas.toNumber() * 1.1,
+            );
+          }
+
+          if (transactionToEdit.maxFeePerGas) {
+            transactionToEdit.maxFeePerGas = new BigNumber(
+              transactionToEdit.maxFeePerGas.toNumber() * 1.3,
+            );
+          }
+        } else {
+          if (transactionToEdit.gasPrice) {
+            transactionToEdit.gasPrice = new BigNumber(
+              transactionToEdit.gasPrice.toNumber() * 1.3,
+            );
+          }
+        }
+
+        setTransaction(
+          bridge.updateTransaction(transaction, transactionToEdit),
+        );
+
+        break;
+
+      case "speedup":
+        transactionToEdit.nonce = operation.transactionSequenceNumber;
+        transactionToEdit.networkInfo = null;
+        transactionToEdit.gasPrice = null;
+        transactionToEdit.maxFeePerGas = null;
+        transactionToEdit.maxPriorityFeePerGas = null;
+        transaction.mode = "send";
+        transactionToEdit.recipient =
+          (account as Account)?.freshAddress ??
+          (parentAccount as Account)?.freshAddress;
+
+        setTransaction(
+          bridge.updateTransaction(transaction, transactionToEdit),
+        );
+
+        break;
+      default:
+        break;
+    }
+  };
+
+  useEffect(() => {
+    // if cancel
+    if (transaction.amount.eq(0)) {
+      navigation.navigate(ScreenName.SendSelectDevice, {
+        accountId: account.id,
+        parentId: parentAccount?.id,
+        transaction,
+        status,
+      });
+      // if speedup
+    } else if (transaction.networkInfo === null) {
+      navigation.navigate(ScreenName.SendSummary, {
+        accountId: account.id,
+        parentId: parentAccount?.id,
+        transaction,
+        transactionRaw: operation.transactionRaw! as TransactionRaw,
```

Comment:
Same thing about non-null assertion here, feels dangerous to me without a proper test before